cmd/dogstatsd/main.go

```diff
@@ -88,26 +88,28 @@ func init() {
 }
 
 func start(cmd *cobra.Command, args []string) error {
+	// Main context passed to components
+	ctx, cancel := context.WithCancel(context.Background())
+
+	defer func(cancel context.CancelFunc) {
```

Comment:
In this particular case, is there a difference with `defer stopAgent(cancel)`?

Reply:
no actually, good catch, since the function arguments are evaluated when the `defer` is evaluated, not when the function is executed. Let me change this 👍